lib/openstudio-standards/utilities/object_info.rb

```diff
@@ -1,20 +1,41 @@
-#
-# Get hash of object field names and values
-# @author: Xuechen (Jerry) Lei, PNNL
-# @param  obj [object]
-#
-# @return [Hash<String>] FieldName:Value
-#
-def getObjectHash(obj)
-  fields_array = obj.to_s.split(/\n/)
-  output_hash = { 'object type' => fields_array.shift.split(/,/)[0] }
-  right = nil
-  fields_array.each do |ori_field|
-    left, right = ori_field.split(/[,;]/)
-    left = left.strip
-    right.slice!('!-')
-    right = right.strip
-    output_hash[right] = left
+Standard.class_eval do
+  # A set of method to manipulate objects
+
+  # Get hash of object field names and values
+  # @author: Xuechen (Jerry) Lei, PNNL
+  # @param  obj [object]
+  #
+  # @return [Hash<String>] FieldName:Value
+  #
+  def model_get_object_hash(obj)
+    fields_array = obj.to_s.split(/\n/)
+    output_hash = { 'object type' => fields_array.shift.split(/,/)[0] }
+    right = nil
+    fields_array.each do |ori_field|
+      left, right = ori_field.split(/[,;]/)
+      left = left.strip
+      right.slice!('!-')
+      right = right.strip
+      output_hash[right] = left
+    end
+    return output_hash
+  end
+
+  # This method is used to automatically cast a model
+  # object based on its IDD object type
+  #
+  # @param [OpenStudio::Model::ModelObject]
+  # @return Casted OpenStudio object or nil if the cast was not possible
+  def model_cast_model_object(model_object)
```

Comment:
I think this method is unnecessary and could result in weird behavior, particularly for HVAC where it is called. I suggest refactoring to remove it in the two places it is used. Happy to think about how to refactor those methods.

Reply:
This method is actually quite convenient. For example, with a few lines of code, it allow us to check if an HVAC related component in a model is autosized without actually knowing what its full object type name is in advance. Without it we would have to list in the code all the object that we want to check. The proposed approach also supports future objects to be included in OpenStudio/EnergyPlus (i.e., new coil objects). It has existed in the codebase for a while now, we just generalized it here to use it in multiple places.

What are the weird behavior that you'd expect to see? If you don't want to expose it we could make it private.